File: QTD.SPA/src/app/_DtoModels/ReportSkeleton/ReportSkeleton.ts
import { Entity } from "../Entity";
import { ReportSkeletonColumn } from "./ReportSkeletonColumn";
import { ReportSkeletonFilter } from "./ReportSkeletonFilter";

export class ReportSkeleton extends Entity {
    defaultTitle: string;
    availableFilters: ReportSkeletonFilter[];
    displayColumns:ReportSkeletonColumn[];
}

File: QTD2.Domain/Entities/Core/ReportSkeleton/ReportSkeleton.cs
using QTD2.Domain.Entities.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Domain.Entities.Core
{
    public class ReportSkeleton: Entity
    {
        public string DefaultTitle { get; set; }
        public virtual List<ReportSkeletonFilter> AvailableFilters { get; set; }
        public virtual List<ReportSkeletonColumn> DisplayColumns { get; set; }

        public ReportSkeletonFilter GetFilter(string name)
        {
            if (AvailableFilters == null) AvailableFilters = new List<ReportSkeletonFilter>();

            return AvailableFilters.Where(r => r.Name.ToUpper() == name.ToUpper()).FirstOrDefault();
        }
    }
}
